### PR TITLE
Create HTTPCheck which checks for a given method against some status code

### PR DESCRIPTION
**Pull request by metalmatze**
This was missing and I want to test against a 301 for example.

/cc @kakkoyun

### CODE REVIEW

File: health/checks.go [hunk 1]
```diff
@@ -40,21 +40,33 @@ func TCPDialCheck(addr string, timeout time.Duration) Check {
...
+		}
+
+		resp, err := client.Do(req)
 		if err != nil {
 			return err
 		}
-		resp.Body.Close()
-		if resp.StatusCode != 200 {
-			return fmt.Errorf("returned status %d", resp.StatusCode)
+		if resp.StatusCode != status {
```
Comment (kakkoyun) [inline health/checks.go:70]: The response body should still need to be closed.

```suggestion
		resp.Body.Close()
		if resp.StatusCode != status {
```

Comment (metalmatze) [reply health/checks.go:70]: We don't read it at all? Are you sure? 🤔

Comment (kakkoyun) [reply health/checks.go:70]: I think so.`If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close.`
https://golang.org/pkg/net/http/#Client.Do

It's a good practice to make sure all underlying resources released.

Comment (metalmatze) [reply health/checks.go:70]: Ok. Good. I am aware but thought that when not reading the body, it's also fine to not defer the close. I will add it to better be safe than sorry. :)

**Pull request merged**